tests/txt.test.js

```diff
@@ -112,6 +112,126 @@ describe('validateCodeComments', () => {
+    await expect(validateCopyrightDate(doc, { mode: MODES.SUBMISSION })).resolves.toHaveLength(0)
+  })
+  test('Copyright console date valid', async () => {
+    const doc = cloneDeep(baseTXTDoc)
+    doc.data.extractedElements.copyrightDates = [2025]
+
+    await expect(validateCopyrightDate(doc, { mode: MODES.NORMAL, year: 2025 })).resolves.toHaveLength(0)
+    await expect(validateCopyrightDate(doc, { mode: MODES.FORGIVE_CHECKLIST, year: 2025 })).resolves.toHaveLength(0)
+    await expect(validateCopyrightDate(doc, { mode: MODES.SUBMISSION, year: 2025 })).resolves.toHaveLength(0)
+  })
+  test('Copyright text date not valid', async () => {
+    const doc = cloneDeep(baseTXTDoc)
+
+    doc.data.extractedElements.copyrightDates = [2023]
+
+    await expect(validateCopyrightDate(doc, { mode: MODES.NORMAL })).resolves.toContainError('COPYRIGHT_DATE_NOT_VALID', ValidationWarning)
+    await expect(validateCopyrightDate(doc, { mode: MODES.FORGIVE_CHECKLIST })).resolves.toContainError('COPYRIGHT_DATE_NOT_VALID', ValidationWarning)
+    await expect(validateCopyrightDate(doc, { mode: MODES.SUBMISSION })).resolves.toContainError('COPYRIGHT_DATE_NOT_VALID', ValidationWarning)
+  })
+  test('Copyright console date not valid', async () => {
+    const doc = cloneDeep(baseTXTDoc)
+
+    doc.data.extractedElements.copyrightDates = [2034]
+
+    await expect(validateCopyrightDate(doc, { mode: MODES.NORMAL, year: 2024 })).resolves.toContainError('COPYRIGHT_DATE_NOT_VALID', ValidationWarning)
+    await expect(validateCopyrightDate(doc, { mode: MODES.FORGIVE_CHECKLIST, year: 2024 })).resolves.toContainError('COPYRIGHT_DATE_NOT_VALID', ValidationWarning)
+    await expect(validateCopyrightDate(doc, { mode: MODES.SUBMISSION, year: 2024 })).resolves.toContainError('COPYRIGHT_DATE_NOT_VALID', ValidationWarning)
+  })
+})
+
+describe('validateLicenseDeclarations', () => {
+  test('should return error when both licence6_b_ii is empty and revisedBsdLicense6_i is false', async () => {
+    const doc = cloneDeep(baseTXTDoc)
+    doc.data.extractedElements.licence6_b_ii = []
+    doc.data.contains.revisedBsdLicense6_i = false
+    doc.data.slug = 'draft-ietf-example'
+
+    const result = await validateLicenseDeclarations(doc)
+    expect(result).toContainEqual(new ValidationError(
+      'TLP4_LICENSE_NOTICE_MISSING',
+      'The document does not contain a required TLP-4 license notice (6.b.i or 6.b.ii).',
+      { ref: 'https://trustee.ietf.org/license-info' }
+    ))
+  })
+
+  test('should not return error for missing licence6_b_ii if revisedBsdLicense6_i is present', async () => {
+    const doc = cloneDeep(baseTXTDoc)
+    doc.data.extractedElements.license6_b_ii = []
+    doc.data.contains.revisedBsdLicense6_i = true
+    doc.data.slug = 'draft-ietf-example'
+
+    const result = await validateLicenseDeclarations(doc)
+    expect(result).toHaveLength(0)
+  })
+
+  test('should return warning for licence6_c_i when slug starts with "draft-ietf-"', async () => {
```

Comment:
Not asking for a change here, but commenting to help inform working in this problem space. This is a chicken-vs-egg problem - we're using the "draft-ietf" prefix here as a _proxy_ for "is this an IETF stream document. When checking TXT files, there's not much more we can do (except possibly try to trigger of whether the intended status is standards track). But when checking XML files, we can look at the stream attribute.